mifospay/src/main/java/org/mifos/mobilewallet/mifospay/data/firebase/api/services/MifosPayMessagingService.kt

```diff
@@ -144,84 +125,73 @@ private void scheduleJob() {
-        Uri defaultSoundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
-        NotificationCompat.Builder notificationBuilder =
-                new NotificationCompat.Builder(this, channelId)
-                        .setSmallIcon(R.drawable.ic_bank)
-                        .setContentTitle(title)
-                        .setContentText(messageBody)
-                        .setAutoCancel(true)
-                        .setSound(defaultSoundUri)
-                        .setContentIntent(pendingIntent);
-
-        NotificationManager notificationManager =
-                (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+    private fun sendNotification(title: String, messageBody: String) {
+        val intent = Intent(this, NotificationActivity::class.java)
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+        val pendingIntent: PendingIntent = PendingIntent.getActivity(
+            this, 0 /* Request code */,
+            intent,
+            PendingIntent.FLAG_ONE_SHOT
+        )
+        val channelId = getString(R.string.app_name)
+        val defaultSoundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
+        val notificationBuilder = NotificationCompat.Builder(this, channelId)
+            .setSmallIcon(R.drawable.ic_bank)
+            .setContentTitle(title)
+            .setContentText(messageBody)
+            .setAutoCancel(true)
+            .setSound(defaultSoundUri)
+            .setContentIntent(pendingIntent)
+        val notificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
 
         // Since android Oreo notification channel is needed.
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            NotificationChannel channel = new NotificationChannel(channelId,
-                    "Channel human readable title",
-                    NotificationManager.IMPORTANCE_DEFAULT);
-            notificationManager.createNotificationChannel(channel);
+            val channel = NotificationChannel(
+                channelId,
+                "Channel human readable title",
+                NotificationManager.IMPORTANCE_DEFAULT
+            )
+            notificationManager.createNotificationChannel(channel)
         }
-
-        notificationManager.notify(0 /* ID of notification */, notificationBuilder.build());
+        notificationManager.notify(0 /* ID of notification */, notificationBuilder.build())
     }
 
     /**
      * Handles notification messages.
      */
-    private void handleNotification(String title, String message) {
-
-    }
+    private fun handleNotification(title: String, message: String) {}
 
     /**
      * Handles data messages.
      */
-    private void handleDataMessage(JSONObject json) {
-        Log.e(TAG, "push json: " + json.toString());
-
+    private fun handleDataMessage(json: JSONObject) {
+        Log.e(TAG, "push json: $json")
```

Comment:
`Log.e` is misleading. Since these are no errors, let's use log.d

Reply:
@devansh-299 please review made changes